packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,540 @@
+  }
+
+  Widget _buildPlainChildFromBuilder(BuildContext context, int index) {
+    // If this cell has already been built since the last time `build()` was
+    // called, return that cached cell widget.
+    if (_cellWidgetCache[index] != null) {
+      return _cellWidgetCache[index];
+    }
+
+    // We do not have a cached version of this cell. Build the cell's widget
+    // and cache it.
+    final Widget cell = widget.plainChildrenBuilder(context, index);
+    _cellWidgetCache[index] = cell;
+
+    // If the cell doesn't exist, return null.
+    if (cell == null) {
+      return null;
+    }
+
+    // Determine if this cell is the last cell in the CupertinoTableView so that
+    // we can report its height for the sake of drawing extra dividers.
+    bool isLastCell;
+    if (widget.plainChildCount != null) {
+      // The child count was provided so we will simply compare cell indices.
+      isLastCell = index == widget.plainChildCount - 1;
+    } else {
+      // No child count was provided, so the only way to know if this is the
+      // last cell is to build the next cell and check for null.
+      //
+      // First, try to find the next cell in the cell widget cache. If it's not
+      // there then create the next cell and cache it.
+      final Widget nextCell = _cellWidgetCache[index + 1] ?? widget.plainChildrenBuilder(context, index + 1);
+      _cellWidgetCache[index + 1] = nextCell;
+
+      isLastCell = nextCell == null;
+    }
+
+    return _decorateCell(cell, isLastCell);
+  }
+
+  // Decorates an individual cell with a divider beneath it. If this cell
+  // [isLastCell] in the table then this decorator also adds a
+  // [_LayoutSizeReport] so that the height of this last cell can be reported
+  // to the extra dividers sliver.
+  Widget _decorateCell(Widget cell, bool isLastCell) {
+    final Widget cellWithDivider = DecoratedBox(
+      decoration: const CupertinoDividerDecoration(),
+      position: DecorationPosition.foreground,
+      child: cell,
+    );
+
+    if (isLastCell) {
+      return _LayoutSizeReport(
+        sizeCallback: (Size size) {
+          _cellHeightController.cellHeight = size.height;
+        },
+        child: cellWithDivider,
+      );
+    } else {
+      return cellWithDivider;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Clear the cell widget cache because our state has changed and therefore
+    // our cells may have changed.
+    _cellWidgetCache.clear();
+
+    // Visual list of cells.
+    final Widget sliverList = SliverList(
+      delegate: SliverChildBuilderDelegate(
+        _cellDecorator,
+        childCount: widget.plainChildCount,
+      ),
+    );
+
+    // Visual list of cells + extra dividers at the bottom.
+    final Widget scrollingList = CustomScrollView(
+      controller: widget.scrollController,
+      slivers: <Widget>[]
+        ..add(sliverList)
+        ..add(CupertinoTableViewExtraDividers(
```

Comment:
related to above. This can be done by the sliver. If we don't, we should probably add a flag so we grouped cases, we don't keep adding this and running its logic if we never need it.

Reply:
Can you clarify what "the sliver" is?  `CupertinoTableViewExtraDividers` is a sliver...